libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -343,11 +343,10 @@ data Pred (fn :: [Type] -> Type -> Type) where
     List (Weighted (Binder fn)) as ->
     Pred fn
   -- if-then-else where the branches depend on the scrutinee.
-  IfElse ::
+  When ::
     HasSpec fn Bool =>
     Term fn Bool ->
     Pred fn ->
-    Pred fn ->
     Pred fn
```

Comment:
This seems to have lost a Pred for the "else" clause,
Perhaps this is more like the Monadic "when" where the missing else clause  silently returns ()
I would assume here, that the missing else clause means No constraints?
Some comments would clarify this.

Reply:
This is on purpose. We have to rewrite `When b (Block [p, q])` to `Block [When b p, When b q]` and if there was an else clause things would start blowing up in annoying ways.